src/psyclone/tests/nemo/transformations/openacc/data_directive_test.py

```diff
@@ -326,10 +329,8 @@ def test_no_data_ref_read(parser):
     schedule = psy.invokes.invoke_list[0].schedule
     acc_trans = TransInfo().get_trans_name('ACCDataTrans')
     acc_trans.apply(schedule.children)
-    with pytest.raises(NotImplementedError) as err:
-        _ = str(psy.gen)
-    assert ("derived-type references on the RHS of assignments are not yet "
-            "supported" in str(err.value))
+    gen_code = str(psy.gen)
+    assert "COPYIN(fld,fld%data)" in gen_code
```

Comment:
Whoop!

Reply:
Yes, that felt really good. I just noticed a change in the read-only detection as well (might be part of #1271, I need to verify this in detail and add more tests, but will do this in 1271) ... now the loop boundaries in gocean are detected as being constants as well, which is nice.